### PR TITLE
Tidy up modules and CI build scripts

### PR DESCRIPTION
**Pull request by VikashKothary**

### CODE REVIEW

File: docs/pom.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    <properties>
+        <docs.site.port>40080</docs.site.port>
+    </properties>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-site-plugin</artifactId>
+                <version>3.7.1</version>
```
Comment (rhys-saldanha) [inline docs/pom.xml]: is this needed? does Spring not manage this plugin version?

File: config/.env.dry [hunk 2]
```diff
@@ -1,17 +0,0 @@
-# .env.dry (reciplease)
```
Comment (rhys-saldanha) [inline config/.env.dry:1]: I disagree with deleting this file, it's basically a way of testing all the commands without actually running them

**Review by rhys-saldanha — CHANGES_REQUESTED** [review]

File: config/.env.dry (hunk 2 above)
Comment (VikashKothary) [reply config/.env.dry:1]: Go for it, create your own using `make config ENV=dry`

Comment (VikashKothary) [reply config/.env.dry:1]: The `.env.example` should give you an idea of what you can customise, and should include sensible defaults.

File: docs/pom.xml (hunk 1 above)
Comment (VikashKothary) [reply docs/pom.xml]: So no. This isn't covered by Spring (as far as I understand it). 

But you're right, the version should be specified in the `parent` module.

File: config/.env.dry (hunk 2 above)
Comment (VikashKothary) [reply config/.env.dry:1]: "Whilst I can understand what you're doing, I'm not the biggest fan" - Vikash, 2021

**Review by rhys-saldanha — APPROVED** [review]

**Pull request merged**